### PR TITLE
Feature/customize confirmation mail

### PR DESCRIPTION
**Pull request by Malte-D**
This also includes "Enable/disable cancel and reschedule links",  because the links can now simply be removed from the template

### CODE REVIEW

File: lib/calendarClient.ts [hunk 1]
```diff
@@ -506,13 +507,17 @@ const listCalendars = (withCredentials) =>
     results.reduce((acc, calendars) => acc.concat(calendars), [])
   );
 
-const createEvent = async (credential, calEvent: CalendarEvent): Promise<any> => {
+const createEvent = async (
+  credential,
+  calEvent: CalendarEvent,
+  emailTemplates: EmailTemplate[]
+): Promise<any> => {
```
Comment (baileypumfleet) [inline lib/calendarClient.ts:514]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689304)
```suggestion
): Promise<unknown> => {
```

**Comment by Malte-D** [comment]
waiting for #330 and #321, after that this will be changed accordingly

**Comment by baileypumfleet** [comment]
Both #330 and #321 have now been merged :)

File: lib/videoClient.ts [hunk 2]
```diff
@@ -107,79 +112,103 @@ const ZoomVideo = (credential): VideoApiAdapter => {
...
+const getBusyVideoTimes = (withCredentials, dateFrom, dateTo) =>
+  Promise.all(videoIntegrations(withCredentials).map((c) => c.getAvailability(dateFrom, dateTo))).then(
+    (results) => results.reduce((acc, availability) => acc.concat(availability), [])
+  );
+
+const createMeeting = async (
+  credential,
+  calEvent: CalendarEvent,
+  emailTemplates: EmailTemplate[]
+): Promise<any> => {
```
Comment (baileypumfleet) [inline lib/videoClient.ts:205]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689304)
```suggestion
): Promise<unknown> => {
```

File: lib/videoClient.ts [hunk 3]
```diff
@@ -192,64 +221,68 @@ const createMeeting = async (credential, calEvent: CalendarEvent): Promise<any>
...
 
   return {
     uid,
-    createdEvent: creationResult
+    createdEvent: creationResult,
   };
 };
 
-const updateMeeting = async (credential, uidToUpdate: String, calEvent: CalendarEvent): Promise<any> => {
+const updateMeeting = async (credential, uidToUpdate: string, calEvent: CalendarEvent): Promise<any> => {
```
Comment (baileypumfleet) [inline lib/videoClient.ts:245]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689304)
```suggestion
const updateMeeting = async (credential, uidToUpdate: string, calEvent: CalendarEvent): Promise<unknown> => {
```

File: lib/videoClient.ts [hunk 4]
```diff
@@ -192,64 +221,68 @@ const createMeeting = async (credential, calEvent: CalendarEvent): Promise<any>
...
 
   return {
     uid: newUid,
-    updatedEvent: updateResult
+    updatedEvent: updateResult,
   };
 };
 
-const deleteMeeting = (credential, uid: String): Promise<any> => {
+const deleteMeeting = (credential, uid: string): Promise<any> => {
```
Comment (baileypumfleet) [inline lib/videoClient.ts:280]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689304)
```suggestion
const deleteMeeting = (credential, uid: string): Promise<unknown> => {
```

**Comment by Malte-D** [comment]
@Nico-J can you check if your changes from #330 are still working correctly?

**Review by Nico-J — APPROVED** [review]
Looks really good to me. Though, I have made some minor annotations to your code that could be improved.

Nevertheless, I don't receive the AttendeeEmails when booking an event. A quick look through the code brings me to line 267 in the calendarClient.ts, where disableConfirmationEmail is set to true for Outlook calendar events. Then, in the booking process, no AttendeeEmail is sent, because the flag mentioned above is true and in addition, we received a creationResult from the API.
Question rather goes to @emrysal : Why do we disable the confirmation mails for Outlook? Do we want to avoid doubled emails (the one from Outlook and the one from us)?

Anyway, when the annotations I made have been considered, take this as approved.

**Comment by baileypumfleet** [comment]
Awesome, once Nico's comments have been resolved and the merge conflicts are resolved, happy to merge this in :)

**Comment by baileypumfleet** [comment]
Is this something you'd be able to complete over the weekend?

**Comment by PeerRich** [comment]
too out of date and thereby too many merge conflicts. also too low priority to be fixed. big big apologies for not merging this early but thank you for your work nevertheless! 🙏

**Pull request closed without merging**